Mathlib/CategoryTheory/Comma/Over/OverClass.lean

```diff
@@ -140,4 +140,44 @@ instance OverClass.fromOver {S : C} (X : Over S) : OverClass X.left S where
 instance {S : C} {X Y : Over S} (f : X ⟶ Y) : HomIsOver f.left S where
   comp_over := Over.w f
 
-end CategoryTheory
+variable [OverClass X S] [OverClass Y S] [OverClass Z S]
+
+/-- Reinterpret an isomorphism over an object `S` into an isomorphism in the category over `S`. -/
+def Iso.asOver (e : X ≅ Y) [HomIsOver e.hom S] : OverClass.asOver X S ≅ OverClass.asOver Y S :=
+  Over.isoMk e (by simp)
+
+@[simp] lemma Iso.asOver_hom (e : X ≅ Y) [HomIsOver e.hom S] :
+    (e.asOver S).hom = OverClass.asOverHom S e.hom := rfl
+
+@[simp] lemma Iso.asOver_inv (e : X ≅ Y) [HomIsOver e.hom S] :
+    have : HomIsOver e.inv S := ⟨by rw [e.inv_comp_eq]; simp⟩
+    (e.asOver S).inv = OverClass.asOverHom S e.inv := rfl
+
```

Comment:
I'd suggest moving this section after the `OverClass` namespace/section. This way, this can be condensed in
```lean
/-- Reinterpret an isomorphism over an object `S` into an isomorphism in the category over `S`. -/
@[simps]
def Iso.asOver (e : X ≅ Y) [HomIsOver e.hom S] : OverClass.asOver X S ≅ OverClass.asOver Y S where
  hom := OverClass.asOverHom S e.hom
  inv := OverClass.asOverHom S e.inv
```

This also makes it so that you don’t have to have a statement with a `have`-expression in `Iso.asOver_inv` (if you want to keep things this way, this should at least be an `haveI`.)